### ISSUE TITLE
Unclear documentation for methods setInAppMessageVisibility() and setLauncherVisibility()

### ISSUE THREAD

**Issue by hanspeide** [intercom/intercom-android#241]
```
    /**
     * <p>Toggle visibility of messages within the Intercom sdk</p>
     *
     * <p>Set as Intercom.GONE to hide all incoming Intercom messages and message previews in the parts of your app
     * where you do not wish to interrupt users: camera views, parts of a game etc.</p>
     *
     * @param visibility Intercom.GONE for hidden, Intercom.VISIBLE to show
     * @since 3.0.0
     */
    public abstract void setInAppMessageVisibility(Visibility visibility);

    /**
     * <p>Toggle visibility of messages within the Intercom sdk</p>
     *
     * <p>Set as Intercom.GONE to hide all incoming Intercom messages and message previews in the parts of your app
     * where you do not wish to interrupt users: camera views, parts of a game etc.</p>
     *
     * @param visibility Intercom.GONE for hidden, Intercom.VISIBLE to show
     * @since 3.0.0
     */
    public abstract void setLauncherVisibility(Visibility visibility);
```

Documentation for these methods are the same, and it's not clear to me what the difference between them is.

**Comment by ronocod** [comment]
Thanks for highlighting this, we'll fix this in our next release.

**Comment by hanspeide** [comment]
Bonus question: How do I prevent the chat window from showing up when in a certain fragment in my app? I tried calling `setInAppMessageVisibility(Intercom.Visibility.VISIBLE)` in `onResume()` and `setInAppMessageVisibility(Intercom.Visibility.GONE)` in `onPause()`, but that doesn't seem to do anything. Bug or feature?

**Comment by ronocod** [comment]
By chat window do you mean the [Messenger window itself](https://marketing.intercomassets.com/assets/_redesign/mv3/use-cases/mv3-native-mobile-3-support-c955028b5efe4e8a85a35f914c9cb057a34f48517a4d904e694504ef0de026ae.jpg) or the in-app messages? Your method should work fine for the in-app ones.

**Comment by hanspeide** [comment]
IAM I guess. The popup shown when receiving a message when in the app.

**Comment by ronocod** [comment]
That seems unusual. Can you set a breakpoint on `setInAppMessageVisibility()` to make sure it's not getting called from anywhere else?

**Comment by hanspeide** [comment]
Looks like the last one was caused by my code not being in sync. My bad.

**Comment by hanspeide** [comment]
Back with another question. Let me know if you want me to open a separate issue if you can reproduce:

Background: I have several fragments in a pager. For one of the fragments I would only like to show the unread counter when there's an incoming message (so I call `setInAppMessageVisibility(Intercom.Visibility.GONE)`), but for the other I want to show the IAM popup (and therefore call `setInAppMessageVisibility(Intercom.Visibility.GONE)`.

Scenario: I'm in the fragment that shows the counter. I receive a message, and open it. After this I go back to the fragment, and I can see that the unread counter is no longer visible, as expected. Now when I swipe to the other fragments the IAM is shown, even though I've already opened and read it.

**Comment by ronocod** [comment]
That's definitely a bug on our end. Could you file a separate issue for it? We'll get to work on it.

**Comment by hanspeide** [comment]
Sure. #243

**Comment by ronocod** [comment]
Great thanks. We'll close this issue when we release the next version.

**Comment by ronocod** [comment]
Fixed in 3.0.4.
